addons/crm/models/crm_lead.py

```diff
@@ -1635,6 +1629,26 @@ def _pls_increment_frequency(self, frequencies, field, value, won, lost):
             frequencies[field][value]['lost'] += lost
         return frequencies
 
+    def _update_lead_probability_batch(self, lead_probabilities, leads_to_update):
+        query = "update crm_lead set automated_probability = %s where id = %s"
+        try:
+            from psycopg2 import extras
+            # Update in execute batch to avoid server roundtrips + page_size to 10000 to avoid memory errors
+            batch_params = [(lead_probabilities[lead.id], lead.id) for lead in leads_to_update if
+                            lead.id in lead_probabilities]
+            extras.execute_batch(self._cr, query, batch_params, page_size=10000)
+        # Fallback in case of psycopg2.7 (minimum) is not installed.
```

Comment:
requirements: psycopg2==2.7.3.1 -> it is really required ?

Reply:
Aaaaand this PR is finally completely useless..